torchquad/integration/gaussian.py

```diff
@@ -1,99 +1,140 @@
+            backend (string): which backend array to return
 
         Returns:
-            float: integral value
+            backend tensor: the weights
+        """
+        weights = anp.array(self._cached_points_and_weights(N)[1], like=backend)
+        if backend == "torch":
+            weights.requires_grad = requires_grad
+            return anp.prod(
+                anp.array(
+                    anp.stack(
+                        list(anp.meshgrid(*([weights] * dim))), like=backend, dim=0
+                    )
+                ),
+                axis=0,
+            ).ravel()
+        else:
+            return anp.prod(
+                anp.meshgrid(*([weights] * dim), like=backend), axis=0
+            ).ravel()
+
+    def _roots(self, N, backend, requires_grad=False):
+        """return the roots generated from the polynomial of choice
 
+        Args:
+            N (int): number of nodes
+            backend (string): which backend array to return
+
+        Returns:
+            backend tensor: the roots
         """
-        if not integration_domain or self.name in ["Gauss-Laguerre","Gauss-Hermite"]:
-            integration_domain=self.default_integration_domain*dim
-            if self.name in ["Gauss-Laguerre","Gauss-Hermite"]:
-                logger.info(f"{self.name} integration only allowed over the interval {self.default_integration_domain[0]}!")
-                
-        self._integration_domain = _setup_integration_domain(dim, integration_domain)
-        self._check_inputs(dim=dim, N=N, integration_domain=self._integration_domain)
-            
-        logger.debug(f"Using {self.name} for integrating a fn with {N} points over {self._integration_domain}")
-
-        self._dim = dim
-        self._fn = fn
-        const=self._get_constant_multiplier()
-        
-        root_args=(N,)+self.root_args
-        
-        xi, wi = self._points_and_weights(self.root_fn,root_args,wrapper_func=self.wrapper_func)
-
-        integral= anp.sum(const*anp.sum(self._eval(xi,args=args,weights=wi),axis=1)) #what if there is a sum in the function? then wi*self._eval() will have dimension mismatch
-        logger.info(f"Computed integral was {integral}.")
-
-        return integral
+        roots = anp.array(self._cached_points_and_weights(N)[0], like=backend)
+        if requires_grad:
+            roots.requires_grad = True
+        return roots
+
+    @property
+    def _grid_func(self):
+        def f(a, b, N, requires_grad, backend=None):
+            return self._resize_roots(a, b, self._roots(N, backend, requires_grad))
+
+        return f
+
+    def _resize_roots(self, a, b, roots):  # scale from [-1,1] to [a,b]
+        """resize the roots based on domain of [a,b]
+
+        Args:
+            a (backend tensor): lower bound
+            b (backend tensor): upper bound
+            roots (backend tensor): polynomial nodes
+
+        Returns:
+            backend tensor: rescaled roots
+        """
+        return roots
```

Comment:
? not sure if intentional? :)

Reply:
Yes, I just put some default behavior here i.e., for integrals over a fixed interval that don't need rescaling as is the case with some of the other Gaussian integrals.